src/item_factory.cpp

```diff
@@ -2184,6 +2184,26 @@ void Item_factory::load_basic_info( JsonObject &jo, itype &def, const std::strin
         def.looks_like = jo.get_string( "looks_like" );
     }
 
+    if( jo.has_member( "conditional_names" ) ) {
+        if( jo.has_array( "conditional_names" ) ) {
```

Comment:
You don't need to check for this, just call `get_array`. The JSON system will automatically generate an error message if the member of that name is not an array.